Make API more generic

NyCodeGHG opened the pull request:
This PR makes the API more generic to also allow the use as a library without having to write to temp files.

sickill commented on src/lib.rs:
```diff
@@ -1,7 +1,10 @@
 use anyhow::{anyhow, Result};
+use asciicast::Header;
 use clap::ArgEnum;
+use gifski::progress::ProgressReporter;
```
Same here, not used?

sickill reviewed (commented):
Looks great, thanks! Just 2 imports I'm not sure about.

NyCodeGHG commented:
Hi @sickill I fixed the imports and a few other things using `cargo clippy --fix`, hope the other changes are not a problem.
I also catched a compilation error and also fixed that so that the cli actually keeps working.

sickill reviewed (approved):
Excellent, thank you. And I like the simplification with `#[default]` :ok_hand:
